templates/search/search_clients_last_seen_v1/query.sql

```diff
@@ -0,0 +1,164 @@
+      COALESCE(ad_click, 0) AS ad_click,
+      COALESCE(search_with_ads, 0) AS search_with_ads,
+      * EXCEPT (ad_click, search_with_ads)
+    FROM
+      search_clients_daily_v7
+    WHERE
+      submission_date_s3 = @submission_date
+      and sample_id = "84"
+  ),
+
+  _derived_engine_searches AS (
+    SELECT
+      STRUCT(
+        short_engine AS key,
+        STRUCT(
+          total_searches,
+          tagged_searches,
+          ad_click,
+          search_with_ads
+        ) AS value
+      ) AS engine_searches,
+      * EXCEPT (short_engine)
+    FROM
+      _derived_search_cols
+  ),
+
+  _grouped AS (
+    -- Here we get a single row per-client, containing
+    -- info from each engine, as well as overall info
+    SELECT
+      -- Grouping columns
+      client_id,
+      CAST(sample_id AS INT64) AS sample_id,
+
+      -- Dimensional data
+      udf_mode_last(ARRAY_AGG(country)) AS country,
+      udf_mode_last(ARRAY_AGG(app_version)) AS app_version,
+      udf_mode_last(ARRAY_AGG(distribution_id)) AS distribution_id,
+      udf_mode_last(ARRAY_AGG(locale)) AS locale,
+      udf_mode_last(ARRAY_AGG(search_cohort)) AS search_cohort,
+      udf_mode_last(ARRAY_AGG(addon_version)) AS addon_version,
+      udf_mode_last(ARRAY_AGG(os)) AS os,
+      udf_mode_last(ARRAY_AGG(channel)) AS channel,
+      udf_mode_last(ARRAY_AGG(profile_creation_date)) AS profile_creation_date,
+      udf_mode_last(ARRAY_AGG(default_search_engine)) AS default_search_engine,
+      udf_mode_last(ARRAY_AGG(default_search_engine_data_load_path)) AS default_search_engine_data_load_path,
+      udf_mode_last(ARRAY_AGG(default_search_engine_data_submission_url)) AS default_search_engine_data_submission_url,
+      udf_mode_last(ARRAY_AGG(profile_age_in_days)) AS profile_age_in_days,
+      udf_mode_last(ARRAY_AGG(active_addons_count_mean)) AS active_addons_count_mean,
+      udf_mode_last(ARRAY_AGG(user_pref_browser_search_region)) AS user_pref_browser_search_region,
+      udf_mode_last(ARRAY_AGG(os_version)) AS os_version,
+
+      -- User activity data
+      MAX(max_concurrent_tab_count_max) AS max_concurrent_tab_count_max,
+      SUM(tab_open_event_count_sum) AS tab_open_event_count_sum,
+      SUM(active_hours_sum) AS active_hours_sum,
+      SUM(subsession_hours_sum) AS subsession_hours_sum,
+      SUM(sessions_started_on_this_day) AS sessions_started_on_this_day,
+
+      -- Search data
+      SUM(organic) AS organic,
+      SUM(sap) AS sap,
+      SUM(unknown) AS unknown,
+      SUM(tagged_sap) AS tagged_sap,
+      SUM(tagged_follow_on) AS tagged_follow_on,
+      SUM(ad_click) AS ad_click,
+      SUM(search_with_ads) AS search_with_ads,
+      SUM(total_searches) AS total_searches,
+      SUM(tagged_searches) AS tagged_searches,
+
+      -- Monthly search totals
+      udf_aggregate_search_map(ARRAY_AGG(engine_searches)) AS engine_searches
+
+    FROM _derived_engine_searches
+    GROUP BY
+      sample_id,
+      client_id
```

Comment:
this data derives from `search_clients_daily`, which does not contain multiple values per `client_id` on a given `submission_date`, so this group by expression will have no effect other than casting `sample_id` to `INT64`

Reply:
nevermind, i'm wrong